src/components/button/button-types.style.js

```diff
@@ -25,6 +27,9 @@ export default ({ colors, disabled }, isDisabled) => ({
         background: ${colors.secondary};
         border-color: ${colors.secondary};
         color: ${colors.white};
+        ${StyledIcon} {
+          color: ${colors.white};
+        }
```

Comment:
I've just unresolved this, because it looks like this issue is still outstanding in the latest commit (3579b13) on the `FE-1691-icon` branch – see https://github.com/Sage/carbon/blob/3579b13/src/components/button/button-types.style.js#L29-L32

Reply:
Many thanks for refactoring this in commit 198047a :white_check_mark: